internal/utils/commands_test.go

```diff
@@ -276,18 +269,21 @@ func TestRunCommand(t *testing.T) {
 	stdout = &bytes.Buffer{}
 
 	// assert background cmd returns process
-	cmd, err = RunCommand(context.TODO(), "", hcmd, "", stdout, stderr, logger, 0, "")
+	cmd, err = RunCommand(t.Context(), "", hcmd, "", stdout, stderr, logger, 0, "")
 	assert.NoError(t, err)
 	assert.NotNil(t, cmd)
-	// no stdout for background processes
-	assert.Empty(t, strings.TrimSpace(stdout.String()))
+
+	// Wait for background process to complete before checking stdout
+	err = cmd.Wait()
+	assert.NoError(t, err)
+	assert.Equal(t, "hello", strings.TrimSpace(stdout.String()))
```

Comment:
Hi @porridge, it seems that there was a race condition in `TestRunCommand` test, since the test is trying to access to stdout result without wait the end of the process (so, some times the unit test pipeline could fail).
I think this change should be enough to test the background run.
What do you think about run unit tests with `-race` option in pipeline?